### PR TITLE
Feature toggle

### PR DESCRIPTION
**Pull request by matiascalegaris**
Add support for feature toggles and some network logs

### CODE REVIEW

File: Codigo/Protocol.bas [hunk 1]
```diff
@@ -19381,9 +19384,17 @@ End Sub
 
 Private Sub HandleDebugRequest(ByVal UserIndex As Integer)
 On Error GoTo HandleDebugRequest_Err:
-    
-    If (UserList(UserIndex).flags.Privilegios And (e_PlayerType.Admin Or e_PlayerType.Dios Or e_PlayerType.Consejero)) Then
-        Call WriteDebugLogResponse(UserIndex)
+    Dim debugType As Byte
+    Dim args() As String
+    debugType = Reader.ReadInt8()
+    If debugType = 1 Then
```
Comment (morgolock) [inline Codigo/Protocol.bas:19390]: if debugType is invalid value then call `exit sub`

File: Codigo/Logging.bas [hunk 2]
```diff
@@ -56,7 +56,7 @@ Private Declare Function RegisterEventSource Lib "advapi32.dll" Alias "RegisterE
  ByVal lpSourceName As String) As Long
 
 
-Public Sub InitializeCircularLogBuffer(Optional size As Integer = 10)
+Public Sub InitializeCircularLogBuffer(Optional size As Integer = 30)
```
Comment (morgolock) [inline Codigo/Logging.bas:59]: byval

File: Codigo/Protocol.bas [hunk 3]
```diff
@@ -19422,3 +19433,20 @@ Private Sub HandleDeleteItem(ByVal UserIndex As Integer)
 HandleDeleteItem_Err:
 102     Call TraceError(Err.Number, Err.Description, "Protocol.HandleDeleteItem", Erl)
 End Sub
+
+Private Sub HandleFeatureToggle(ByVal UserIndex As Integer)
+On Error GoTo HandleFeatureToggle_Err:
+    Dim value As Byte
+    Dim name As String
+    value = Reader.ReadInt8
+    name = Reader.ReadString8
```
Comment (morgolock) [inline Codigo/Protocol.bas:19442]: if name is empty or too big exit sub before using it

File: Codigo/Protocol.bas [hunk 4]
```diff
@@ -19422,3 +19433,20 @@ Private Sub HandleDeleteItem(ByVal UserIndex As Integer)
 HandleDeleteItem_Err:
 102     Call TraceError(Err.Number, Err.Description, "Protocol.HandleDeleteItem", Erl)
 End Sub
+
+Private Sub HandleFeatureToggle(ByVal UserIndex As Integer)
+On Error GoTo HandleFeatureToggle_Err:
+    Dim value As Byte
+    Dim name As String
+    value = Reader.ReadInt8
```
Comment (morgolock) [inline Codigo/Protocol.bas:19441]: `value = max(min(MaxValue,Reader.ReadInt8),MinValue)`

clamps the values to valid range

**Review by morgolock — CHANGES_REQUESTED** [review]

File: Codigo/Protocol.bas (hunk 1 above)
Comment (matiascalegaris) [reply Codigo/Protocol.bas:19390]: Ill rebase, this should not be on this mr

File: Codigo/FileIO.bas [hunk 5]
```diff
@@ -3291,3 +3330,18 @@ Public Sub CargarDonadores()
             Next i
         End If
 End Sub
+
+Public Function IsFeatureEnabled(featureName As String)
```
Comment (morgolock) [inline Codigo/FileIO.bas:3334]: `byval featureName As String`

File: Codigo/FileIO.bas [hunk 6]
```diff
@@ -3291,3 +3330,18 @@ Public Sub CargarDonadores()
...
+    If FeatureToggles.Exists(featureName) Then
+        IsFeatureEnabled = FeatureToggles.Item(featureName)
+    Else
+        IsFeatureEnabled = False
+    End If
+End Function
+
+Public Sub SetFeatureToggle(name As String, state As Boolean)
```
Comment (morgolock) [inline Codigo/FileIO.bas:3342]: byval both params

**Review by morgolock — CHANGES_REQUESTED** [review]

**Review by morgolock — APPROVED** [review]

**Pull request merged**